src/main/java/com/aws/greengrass/componentmanager/Unarchiver.java

```diff
@@ -50,12 +51,13 @@ static void unzip(File zipFile, File destDir) throws IOException {
                     Utils.createPaths(newFile.getParentFile().toPath());
                     // Only unarchive when the destination file doesn't exist or the file sizes don't match
                     if (!newFile.exists() || zipEntry.getSize() != newFile.length()) {
-                        try (OutputStream fos = Files.newOutputStream(newFile.toPath(),
-                                StandardOpenOption.CREATE,
-                                StandardOpenOption.TRUNCATE_EXISTING,
-                                StandardOpenOption.SYNC);
-                             InputStream is = zf.getInputStream(zipEntry)) {
+                        try (FileChannel fc = FileChannel.open(newFile.toPath(), StandardOpenOption.CREATE,
```

Comment:
need to add StandardOpenOption.WRITE here I think. newOutputStream adds it by default, but FileChannel doesn't.

Reply:
fixed it.